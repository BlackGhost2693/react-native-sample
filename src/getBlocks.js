// @flow

import React from 'react';

import Paragraph from './components/Paragraph';

function getBlocks(bodyData: Object = {}, customStyles: Object = {}): ?React$Element<*> {
  if (!bodyData.blocks) {
    return null;
  }

  return bodyData.blocks
    .map((item: Object): any => {
      switch (item.type) {
        case 'unstyled':
        // case 'paragraph':
          return <Paragraph key={item.key} text={item.text} customStyle={customStyles.unstyled} />;

        default:
          return null;
      }
    });
}

module.exports = getBlocks;
