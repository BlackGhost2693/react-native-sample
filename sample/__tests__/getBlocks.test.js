/* eslint-env jest */
import getBlocks from '../../src/getBlocks';

jest.mock('../../src/components/DraftJsText', () => 'DraftJsText');
jest.mock('../../src/components/BlockQuote', () => 'BlockQuote');
jest.mock('../../src/components/UnorderedListItem', () => 'UnorderedListItem');
jest.mock('../../src/components/OrderedListItem', () => 'OrderedListItem');

describe('return specific component based on type', () => {
  it('DraftJsText when type unstyled', () => {
    const bodyData = { blocks: [{ type: 'unstyled' }] };
    const result = getBlocks(bodyData);
    expect(result[0].type).toBe('DraftJsText');
  });

  it('BlockQuote when type blockquote', () => {
    const bodyData = { blocks: [{ type: 'blockquote', text: 'My text' }] };
    const result = getBlocks(bodyData);
    expect(result[0].type).toBe('BlockQuote');
  });

  it('UnorderedListItem when type unordered-list-item', () => {
    const bodyData = { blocks: [{ type: 'unordered-list-item' }] };
    const result = getBlocks(bodyData);
    expect(result[0].type).toBe('UnorderedListItem');
  });

  it('OrderedListItem when type ordered-list-item', () => {
    const bodyData = { blocks: [{ type: 'ordered-list-item' }] };
    const result = getBlocks(bodyData);
    expect(result[0].type).toBe('OrderedListItem');
  });

  it('atomicHandler function when type atomic', () => {
    const bodyData = { blocks: [{ type: 'atomic' }] };
    const atomicHandler = item => item;
    const result = getBlocks(bodyData, {}, atomicHandler);
    expect(result[0].type).toBe('atomic');
  });

  it('array of null when type is invalid', () => {
    const bodyData = { blocks: [{ type: 'my-own-type' }] };
    const result = getBlocks(bodyData);
    expect(result[0]).toBe(null);
  });

  it('null when bodyData doesn\'t have blocks', () => {
    const bodyData = { data: 'other-data' };
    const result = getBlocks(bodyData);
    expect(result).toBe(null);
  });
});
