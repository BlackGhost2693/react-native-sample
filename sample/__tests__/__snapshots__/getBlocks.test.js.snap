// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewAfterList should pass props 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
/>
`;
